pkg/querier/worker/scheduler_processor.go

```diff
@@ -110,13 +111,17 @@ func (sp *schedulerProcessor) processQueriesOnSingleStream(workerCtx context.Con
 		}
 
 		if err != nil {
-			level.Error(sp.log).Log("msg", "error contacting scheduler", "err", err, "addr", address)
+			level.Warn(sp.log).Log("msg", "error contacting scheduler", "err", err, "addr", address)
 			backoff.Wait()
 			continue
 		}
 
 		if err := sp.querierLoop(c, address, inflightQuery); err != nil {
-			level.Error(sp.log).Log("msg", "error processing requests from scheduler", "err", err, "addr", address)
+			// Do not log an error is the query-scheduler is shutting down.
+			if s, ok := status.FromError(err); !ok || s.Message() != schedulerpb.ErrSchedulerIsNotRunning.Error() {
```

Comment:
nit: would a `strings.Contains` here be less fragile? I'm not sure exactly where this error comes from so it's fine if the answer is "no".

Reply:
As of today there would be no difference because the scheduler returns `schedulerpb.ErrSchedulerIsNotRunning.Error()`. However, if in the future that error it will be wrapped, then it's less fragile using `strings.Contains()`.

I do the change.